samza-api/src/main/java/org/apache/samza/scheduling/Scheduler.java

```diff
@@ -0,0 +1,28 @@
+package org.apache.samza.scheduling;
+
+import org.apache.samza.task.TimerCallback;
+
+
+/**
+ * Provides a way for applications to register some logic to be executed at a future time.
+ */
+public interface Scheduler {
+  /**
+   * Register a keyed callback of {@link TimerCallback} in this task.
+   * The callback will be invoked exclusively with any other operations for this task, e.g. processing, windowing, and
```

Comment:
nit: be invoked periodically, exclusive to any other operations for this task,

Reply:
A single call to scheduleCallback will only schedule a callback to run at one future time and not multiple times, so in my opinion, we don't want to use the term "periodically" here. Would that be ok to leave it out?